json/json_test.go

```diff
@@ -1378,3 +1378,44 @@ func TestGithubIssue18(t *testing.T) {
 		}
 	}
 }
+
+func TestGithubIssue23(t *testing.T) {
+	t.Run("marshal-1", func(t *testing.T) {
+		type d struct{ S map[string]string }
+
+		b, _ := Marshal(map[string]d{"1": {S: map[string]string{"2": "3"}}})
+		if string(b) != `{"1":{"S":{"2":"3"}}}` {
+			t.Error(string(b))
+		}
+	})
+
+	t.Run("marshal-2", func(t *testing.T) {
+		type testInner struct {
+			InnerMap map[string]string `json:"inner_map"`
+		}
+
+		type testOuter struct {
+			OuterMap map[string]testInner `json:"outer_map"`
+		}
+
+		b, _ := Marshal(testOuter{
+			OuterMap: map[string]testInner{
+				"outer": {
+					InnerMap: map[string]string{"inner": "value"},
```

Comment:
Should we also add a test for a triple nested map, while we're here?

Reply:
Sure I can add this 👍